magiccube/cube_print.py

```diff
@@ -76,4 +79,11 @@ def print_cube(self):
 
         # BOTTOM
         result += self._print_top_down_face(cube, Face.D)
+
+        if orientation:
+            cube.rotate([orientation])
+            if cube._store_history:
+                cube._history.pop()
+                cube._history.pop()
```

Comment:
There is a new `undo` method, which seems helpful here.
```suggestion
            cube.undo()
```

Reply:
Perfect, it's implemented !